recipes/runtimeqml/all/test_package/conanfile.py

```diff
@@ -0,0 +1,21 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake
+import os
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+
+    generators = "CMakeDeps", "CMakeToolchain"
+
+    def configure(self):
+        qt = self.dependencies["qt"]
+        qt.options.qtdeclarative = True
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+    
+    def test(self):
+        bin_path = os.path.join("bin", "RuntimeQmlTest")
+        self.run(bin_path, run_environment=True)
```

Comment:
```suggestion
from conan import ConanFile
from conan.tools.build import can_run
from conan.tools.cmake import cmake_layout, CMake
import os


class TestPackageConan(ConanFile):
    settings = "os", "arch", "compiler", "build_type"
    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
    test_type = "explicit"

    def requirements(self):
        self.requires(self.tested_reference_str)

    def layout(self):
        cmake_layout(self)

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def test(self):
        if can_run(self):
            bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
            self.run(bin_path, env="conanrun")
```
Please, follow the template: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/test_package/conanfile.py

All dependencies options should be solved by the package's recipe, not in test package.